### ISSUE TITLE
can't fork empty repo

### ISSUE THREAD

**Issue by ecwyne** [gramps-graphql/gramps-cli#1]
@jlengstorf Unfortunately I can't fork this repo while it's empty nor do I have write privileges so I can't push to a non-master branch either.

**Comment by jlengstorf** [comment]
Sorry about that. Just gave you write access.

**Comment by ecwyne** [comment]
The repo looks a bit funky because there's no `master` branch (yet)

## To Run
```
$ git clone https://github.com/gramps-graphql/gramps-cli
$ cd gramps-cli
$ git checkout ewyne-setup
$ npm i -g .
$ cd ~/Desktop
$ gramps create XKCD
```

**Comment by ecwyne** [comment]
@jlengstorf also take a look at https://github.com/gramps-graphql/gramps-cli/commit/87143a43ec6522f541b415890828680ef1ecaa58.

I can't fully test `gramps start [path/to/data/source] -l/-m` because `@gramps/gramps` isn't ready

**Comment by jlengstorf** [comment]
First, thanks for putting this together!

Looking at this, though, I have a couple concerns.

1. There are multiple copies of the data source base, which seems like a potential nightmare for maintenance — how do we keep these in sync? How can we be sure they're all up to date?
2. The non-REST data source references a connector, but it's not clear where that comes from.

Inside IBM, we had a Yeoman generator for building new µ-services, and it pretty much immediately went out of date because no one would take the time to update all the various templates. I worry that this is already a little more complex than I'd prefer (three branches of templates to keep track of/update), and we haven't even started looking at database patterns yet.

It may be worth holding off on a `create` command until we've built a few DB-backed data sources and seen what we'll actually need. My hope is that we can contain all of the API changes in the connector itself, which means that the entire data source is otherwise identical.

So basically, I guess what I'm saying is that we may be getting a little ahead of ourselves by adding this, and should maybe pause on the CLI until we've nailed down the 1.x API and gotten everything released and stable.

Does that seem reasonable?

**Comment by ecwyne** [comment]
First, can you make `master` the default branch? I'm not able to do so and everything got all wonky when there wasn't a master branch initially.

I agree that having multiple places for `data-source-base` is a nightmare and should be avoided at all costs. I do picture the `gramps create` command becoming the de-facto source of truth for creating new data sources (but I'll bow to your decision on that one).

The current implementation of `gramps create` is more of a proof of concept than a full-featured releasable artifact.

>and should maybe pause on the CLI until we've nailed down the 1.x API and gotten everything released and stable.

The only problem I see with this is that releasing a 1.x API means removing the cli capabilities of `@gramps/gramps`, so there is a bit of a chicken and egg problem.

**Comment by jlengstorf** [comment]
Updated `master` to be default. I'll put in the README soon (#2).

Maybe what we can do is move the `gramps start` command into the CLI, but keep the `create` command on a branch for now — at least until we figure out more details about what a non-REST data source actually looks like.

I agree with you that the CLI should be the source of truth, and after a conversation with @joshblack I think there are some things we can do to make this easier for us to update as we go:

- We can set up `@gramps/data-source-base` to be a template that would be compatible with `node-plop` (the docs will explain how to manually install a data source, but strongly encourage using the CLI)
- `@gramps/cli` will pull the latest version of `@gramps/data-source-base` before building to ensure the latest version is always used
- `create-gramps-data-source` will be a simple wrapper for `@gramps/cli`'s `gramps create [name]` to support `yarn create gramps-data-source [name]`
- The docs will be written up to show the `package.json` install/usage pattern, but global installs will be supported

My hope is that we'll end up with something like this:
- A data source base template (namespace, schema, resolvers, mocks, model)
- Connector templates (REST, Firebase, MongoDB, etc.)

The connector templates can live in their own repos, and will potentially share a common base if we see duplicate code as we're building them.

So, to condense that down to a short list of action steps:

1. Build the CLI to support the existing GrAMPS development command
2. Finalize the GrAMPS API for data sources and the core
3. Build working DB-backed data sources
4. Revisit the `gramps create` command

Are you cool with that as an order of operations?

**Comment by ecwyne** [comment]
Yes, I think having the `gramps create` command in it's own branch and focusing on `gramps start` in master is the right way forward. (That's how I would have liked to have done so from the beginning)

I also think that supporting `yarn create gramps-data-source` is a super interesting idea as well

If you'd like, I can remove all `gramps create` functionality from master and put it in a separate branch.

**Comment by ecwyne** [comment]
I'll also take care of the readme while I separate `master` and `gramps-create` branches unless you say otherwise.

**Comment by ecwyne** [comment]
There are now [two branches](https://github.com/gramps-graphql/gramps-cli/branches) and master does not include any `gramps create` code
